x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/parser/StatementParserTests.java

```diff
@@ -416,19 +416,210 @@ public void testStringAsLookupIndexPattern() {
+                expectInvalidIndexNameErrorWithLineNumber(command, clusterString + "--indexpattern", lineNumber, "-indexpattern");
+                expectInvalidIndexNameErrorWithLineNumber(
+                    command,
+                    clusterString + "\"<--logstash-{now/M{yyyy.MM}}>\"",
+                    lineNumber,
+                    "-logstash-"
+                );
+                expectInvalidIndexNameErrorWithLineNumber(
+                    command,
+                    clusterString + "--<logstash-{now/M{yyyy.MM}}>",
+                    lineNumber,
+                    "-<logstash-{now/M{yyyy.MM}}>"
+                );
+                expectInvalidIndexNameErrorWithLineNumber(command, clusterString + "\"<logstash#{now/d}>\"", lineNumber, "logstash#");
+                expectInvalidIndexNameErrorWithLineNumber(command, clusterString + "<logstash#{now/d}>", lineNumber, "logstash#");
+                expectInvalidIndexNameErrorWithLineNumber(
+                    command,
+                    clusterString + "\"+<logstash-{now/d}>\"",
+                    lineNumber,
+                    "+<logstash-{now/d}>"
+                );
+                expectInvalidIndexNameErrorWithLineNumber(
+                    command,
+                    clusterString + "+<logstash-{now/d}>",
+                    lineNumber,
+                    "+<logstash-{now/d}>"
+                );
+                expectInvalidIndexNameErrorWithLineNumber(
+                    command,
+                    clusterString + "\"_<logstash-{now/d}>\"",
+                    lineNumber,
+                    "_<logstash-{now/d}>"
+                );
+                expectInvalidIndexNameErrorWithLineNumber(
+                    command,
+                    clusterString + "_<logstash-{now/d}>",
+                    lineNumber,
+                    "_<logstash-{now/d}>"
+                );
+                expectInvalidIndexNameErrorWithLineNumber(command, clusterString + "\"<<logstash{now/d}>>\"", lineNumber, "<logstash");
+                expectInvalidIndexNameErrorWithLineNumber(command, clusterString + "<<logstash{now/d}>>", lineNumber, "<logstash");
+                expectInvalidIndexNameErrorWithLineNumber(command, clusterString + "\"<<logstash<{now/d}>>>\"", lineNumber, "<logstash<");
+                expectInvalidIndexNameErrorWithLineNumber(command, clusterString + "<<logstash<{now/d}>>>", lineNumber, "<logstash<");
+                expectInvalidIndexNameErrorWithLineNumber(
+                    command,
+                    clusterString + "\"-<logstash- {now/d{yyyy.MM.dd|+12:00}}>\"",
+                    lineNumber,
+                    "logstash- "
+                );
+            }
+        }
+
+        // comma separated indices
+        // These fail because there is an invalid index name, and there is no index name with wildcard before it
```

Comment:
Is there any test for `index,-index` and/or just `-index`?

Reply:
I thought about it, it is a good point. I'll add tests for `index,-index`, I'll leave it to the `IndicesOptions` to decide whether to error out or not.